# Performance of AF2 and ESMFold as disorder predictors for de novo proteins and random sequences

The CSV files in the `Data` folder contain the data used to generate the figures in the paper. A detailed description of datasetes features is provided below:

### *De novo* proteins

The `de_novo_proteins_data.csv` file contains the data for the *de novo* proteins. The columns are:
- `id` The id of the protein
- `max_pLDDT` The highest pLDDT score of the all 5 models generated by AlphaFold
- `avg_pLDDT` The average pLDDT score of the all 5 models generated by AlphaFold
- `sequence` The amino acid sequence of the protein
- `FlyBaseID` The FlyBase ID of the protein
- `mechanism`The mechanism of emergence of the protein as identified by[Heames et al. (2020)](https://link.springer.com/article/10.1007/s00239-020-09939-z)
- `age` The age of the protein as identified by [Heames et al. (2020)](https://link.springer.com/article/10.1007/s00239-020-09939-z)
- `species` The species of which the protein derives from as identified by [Heames et al. (2020)](https://link.springer.com/article/10.1007/s00239-020-09939-z)
- `seqLen` The length of the protein sequence
- `mean_disorder_flDPnn` The mean disorder score of the protein as predicted by flDPnn
- `fraction_disorder_flDPnn`The fraction of disordered residues of the protein as predicted by flDPnn
- `median_disorder_flDPnn` The median disorder score of the protein as predicted by flDPnn
- `alpha_content_AF`The fraction of residues in an AlphaFold model which are annotated as alpha-helices by DSSP
- `beta_content_AF` The fraction of residues in an AlphaFold model which are annotated as beta-strands by DSSP
- `alpha_content_SPIDER3S` The fraction of residues which are predicted to be in an alpha-helix by SPIDER3-Single
- `beta_content_SPIDER3S` The fraction of residues which are predicted to be in a beta-strand by SPIDER3-Single
- `secondary_structure_proportion_AF`The combined fraction of residues in an AlphaFold model which are annotated as alpha-helices or beta-strands by DSSP
- `fraction_disorder_Iupred3` The fraction of disordered residues of the protein as predicted by IUPRED3
- `pLDDT_ESM` The pLDDT score of the protein model predicted by ESMFold
- `alpha_content_ESM` The fraction of residues in an ESMFold model which are annotated as alpha-helices by DSSP
- `beta_content_ESM` The fraction of residues in an ESMFold model which are annotated as beta-strands by DSSP
- `TMScore_AF2_ESM` The TM-score of the model predicted by AF2 when aligned to the model predicted by ESMFold
- `lddt_AF2_ESM` The lddt score of the model predicted by AF2 when aligned to the model predicted by ESMFold

### Random sequences

The `random_sequences_data.csv` file contains the data for the random sequences. The columns are:
- `id` The id of the protein
- `max_pLDDT` The highest pLDDT score of the all 5 models generated by AlphaFold
- `avg_pLDDT` The average pLDDT score of the all 5 models generated by AlphaFold
- `sequence` The amino acid sequence of the protein
- `seqLen` The length of the protein sequence
- `mean_disorder_flDPnn` The mean disorder score of the protein as predicted by flDPnn
- `fraction_disorder_flDPnn`The fraction of disordered residues of the protein as predicted by flDPnn
- `median_disorder_flDPnn` The median disorder score of the protein as predicted by flDPnn
- `alpha_content_AF`The fraction of residues in an AlphaFold model which are annotated as alpha-helices by DSSP
- `beta_content_AF` The fraction of residues in an AlphaFold model which are annotated as beta-strands by DSSP
- `alpha_content_SPIDER3S` The fraction of residues which are predicted to be in an alpha-helix by SPIDER3-Single
- `beta_content_SPIDER3S` The fraction of residues which are predicted to be in a beta-strand by SPIDER3-Single
- `fraction_disorder_Iupred3` The fraction of disordered residues of the protein as predicted by IUPRED3
- `pLDDT_ESM` The pLDDT score of the protein model predicted by ESMFold
- `alpha_content_ESM` The fraction of residues in an ESMFold model which are annotated as alpha-helices by DSSP
- `beta_content_ESM` The fraction of residues in an ESMFold model which are annotated as beta-strands by DSSP
- `TMScore_AF2_ESM` The TM-score of the model predicted by AF2 when aligned to the model predicted by ESMFold
- `lddt_AF2_ESM` The lddt score of the model predicted by AF2 when aligned to the model predicted by ESMFold

### Conserved proteins

The `established_proteins_data.csv` file contains the data for the conserved proteins. The columns are:
- `uniprotID` The UniProt ID of the protein
- `seqLen` The length of the protein sequence
- `pLDDT` The pLDDT score of the protein model predicted by AlphaFold retrieved from the AlphaFold database
- `sequence` The amino acid sequence of the protein
- `alpha_content_AF`The fraction of residues in an AlphaFold model which are annotated as alpha-helices by DSSP
- `beta_content_AF` The fraction of residues in an AlphaFold model which are annotated as beta-strands by DSSP
- `alpha_content_SPIDER3S` The fraction of residues which are predicted to be in an alpha-helix by SPIDER3-Single
- `beta_content_SPIDER3S` The fraction of residues which are predicted to be in a beta-strand by SPIDER3-Single
- `mean_disorder_flDPnn` The mean disorder score of the protein as predicted by flDPnn
- `fraction_disorder_flDPnn`The fraction of disordered residues of the protein as predicted by flDPnn
- `median_disorder_flDPnn` The median disorder score of the protein as predicted by flDPnn
- `fraction_disorder_Iupred3` The fraction of disordered residues of the protein as predicted by IUPRED3
- `isDeNovo` Whether the protein is *de novo* or not. Some proteins retrieved from the AlphaFold database are also found in the *de novo* dataset. These are annotated as `True` in this column, and were removed from the evalutions.
- `pLDDT_ESM` The pLDDT score of the protein model predicted by ESMFold
- `alpha_content_ESM` The fraction of residues in an ESMFold model which are annotated as alpha-helices by DSSP
- `beta_content_ESM` The fraction of residues in an ESMFold model which are annotated as beta-strands by DSSP
- `TMScore_AF2_ESM` The TM-score of the model predicted by AF2 when aligned to the model predicted by ESMFold
- `lddt_AF2_ESM` The lddt score of the model predicted by AF2 when aligned to the model predicted by ESMFold
- `id` The id of the protein

### DisProt data

The `disprot_full_dataset.csv` file contains the data for the proteins from the DisProt database. The columns are:
- `uniProtID` The UniProt ID of the protein
- `seqLen` The length of the protein sequence
- `pLDDT` The pLDDT score of the protein model predicted by AlphaFold retrieved from the AlphaFold database
- `sequence` The amino acid sequence of the protein
- `alpha_content_AF`The fraction of residues in an AlphaFold model which are annotated as alpha-helices by DSSP
- `beta_content_AF` The fraction of residues in an AlphaFold model which are annotated as beta-strands by DSSP
- `mean_disorder_flDPnn` The mean disorder score of the protein as predicted by flDPnn
- `fraction_disorder_flDPnn`The fraction of disordered residues of the protein as predicted by flDPnn
- `median_disorder_flDPnn` The median disorder score of the protein as predicted by flDPnn
- `fraction_disorder_Iupred3` The fraction of disordered residues of the protein as predicted by IUPRED3

----------------

### Per-residue data

The data of the per-residue evaluations are stored in the files `per_residue_disorder_MSA_pLDDT_deNovo.csv`, `per_residue_disorder_MSA_pLDDT_randomSequences.csv`, and `per_residue_disorder_MSA_pLDDT_conservedProteins.csv`. Each row represnts one amino acid of a respective protein.

The columns are:
- `id` The id of the protein
- `MSA_Depth` The number of non-gap residues in the MSA at a respective position of the sequences
- `pLDDT` The pLDDT score at the respective position of the protein model predicted by AlphaFold
- `disorder_flDPnn` The disorder score at the respective position of the protein as predicted by flDPnn
- `MSA_log` The base 10 logarithm of the MSA depth at the respective position of the protein
- `DSSP` DSSP annotation of the amino acid at the respective position of the protein
- `length_quartile`The assigned lenght quartile of the protein



